### PR TITLE
c-blosc: conan v2 support

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/c-blosc/all/conanfile.py [hunk 1]
```diff
@@ -109,6 +109,13 @@ def generate(self):
     def _patch_sources(self):
         apply_conandata_patches(self)
         rmdir(self, os.path.join(self.source_folder, "cmake"))
+        # Do not install system libs
+        replace_in_file(
+            self,
+            os.path.join(self.source_folder, "CMakeLists.txt"),
+            "include(InstallRequiredSystemLibraries)",
```
Comment (jwillikers) [inline recipes/c-blosc/all/conanfile.py:116]: Can you disable this by setting the CMake variable `CMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP` to true? Docs [here](https://cmake.org/cmake/help/latest/module/InstallRequiredSystemLibraries.html#installrequiredsystemlibraries).

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/c-blosc/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/c-blosc/all/conanfile.py:116]: I guess yes

**Review by jwillikers — APPROVED** [review]

File: recipes/c-blosc/all/conanfile.py [hunk 2]
```diff
@@ -77,58 +74,60 @@ def requirements(self):
...
     def source(self):
-        tools.files.get(self, **self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version],
+            destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BLOSC_INSTALL"] = True
+        tc.variables["BUILD_STATIC"] = not self.options.shared
+        tc.variables["BUILD_SHARED"] = self.options.shared
```
Comment (prince-chrismc) [inline recipes/c-blosc/all/conanfile.py:84]: So close to the right one yet so far https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**